lib/publiccloud/ec2.pm

```diff
@@ -130,6 +130,12 @@ sub upload_img {
     my $vpc_subnet = get_var('PUBLIC_CLOUD_EC2_UPLOAD_VPCSUBNET');
     my $instance_type = get_required_var('PUBLIC_CLOUD_EC2_UPLOAD_INSTANCE_TYPE');
 
+    if (!$vpc_subnet) {
+        my $vpc_id = script_output("aws ec2 describe-vpcs --filters 'Name=tag:Name,Values=tf-vpc' | jq -r '.Vpcs[0].VpcId'");
```

Comment:
Isn't variable scope local to `if`? We'd define it in the body.

Reply:
> Isn't variable scope local to `if`? We'd define it in the body.

This variable is only used in the if block.